src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs

```diff
@@ -0,0 +1,298 @@
+            {
+                MyDictionary.Remove(""Key"");
+            }
+            else
+            {
+                throw new Exception(""Key doesn't exist"");
+            }
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task NegatedCondition_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (!MyDictionary.ContainsKey(""Key""))
+                MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task AdditionalCondition_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (MyDictionary.ContainsKey(""Key"") && MyDictionary.Count > 2)
+                MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task ConditionInVariable_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            var result = MyDictionary.ContainsKey(""Key"");
+            if (result)
+	            MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task RemoveInSeparateLine_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (MyDictionary.ContainsKey(""Key""))
+	            _ = MyDictionary.Count;
+	        MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task AdditionalStatements_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (MyDictionary.ContainsKey(""Key""))
+            {
+                MyDictionary.Remove(""Key"");
```

Comment:
Could we warn in this case too? without suggesting a fixer

Reply:
Done